### PR TITLE
prov/verbs: Delay device initialization

### PR DESCRIPTION
**Pull request by j-xiong**
Address #8962

### CODE REVIEW

File: prov/verbs/src/verbs_init.c [hunk 1]
```diff
@@ -784,12 +784,8 @@ VERBS_INI
 	ofi_hmem_init();
 	ofi_monitors_init();
 #endif
-
-	vrb_os_mem_support(&vrb_gl_data.peer_mem_support,
-			   &vrb_gl_data.dmabuf_support);
-
-	if (vrb_read_params() || vrb_os_ini() ||
-	    vrb_init_info(&vrb_util_prov.info))
+	if (vrb_os_ini())
```
Comment (shefty) [inline prov/verbs/src/verbs_init.c:787]: It looks odd to have os initialization in vrb_read_params.  Maybe combine vrb_os_ini() with vrb_os_mem_support()?

Comment (j-xiong) [reply prov/verbs/src/verbs_init.c:787]: The way github shows the changes is misleading due to the use of macro as the function name. This call is not inside vrb_read_params(), it's inside VERBS_INI.

Comment (shefty) [reply prov/verbs/src/verbs_init.c:787]: You're right.  It's more obvious when I look at the combined changes versus commit by commit.

**Review by shefty — APPROVED** [review]

File: prov/verbs/src/verbs_info.c [hunk 2]
```diff
@@ -1333,15 +1334,29 @@ static int vrb_device_has_ipoib_addr(const char *dev_name)
...
 	struct ibv_context **ctx_list;
 	struct fi_info *fi = NULL, *tail = NULL;
 	const struct verbs_ep_domain *ep_type[VERBS_NUM_DOMAIN_TYPES];
 	int ret = 0, i, j, num_devices, dom_count = 0;
+	static bool initialized = false;
 
+	if (initialized)
+		return 0;
+
+	initialized = true;
```
[the remark this answers is not in the document]
Comment (j-xiong) [reply prov/verbs/src/verbs_info.c:1348]: I thought about that. Do we expect multithreaded app making the first getinfo() calls concurrently? If so, we can add a mutex here.

[the remark this answers is not in the document]
Comment (shefty) [reply prov/verbs/src/verbs_info.c:1348]: The multi-threaded apps I know about are storage based servers.  I think they may call concurrently, but I don't know.  DDN maybe not, but DAOS could.

[the remark this answers is not in the document]
Comment (j-xiong) [reply prov/verbs/src/verbs_info.c:1350]: Updated to add a mutex.

**Review by shefty — APPROVED** [review]

File: prov/verbs/src/verbs_info.c (hunk 2 above)
[the remark this answers is not in the document]
Comment (sydidelot) [reply prov/verbs/src/verbs_info.c:1350]: Thanks for the mutex. DDN effectively calls getinfo() concurrently :+1:

[the remark this answers is not in the document]
Comment (shefty) [reply prov/verbs/src/verbs_info.c:1350]: @sydidelot - thanks for the info

**Pull request merged**